src/Containers/MinimalContainers/tests/test_ConstantSizeMatrix.cpp

```diff
@@ -28,8 +28,16 @@ TEST_CASE("ConstantSizeMatrix basics", "[containers]")
   CHECK(cmat.capacity() == 32);
 
   CHECK_THROWS(cmat.resize(1, 33));
+  CHECK_THROWS(cmat.resize(34, 2));
 
-  CHECK_THROWS(cmat.resize(2, 9));
+  CHECK_NOTHROW(cmat.resize(2, 9));
+  CHECK(cmat.capacity() == 32);
+
+  // If the new dimensions are not commensurable with the old ones,
+  // the capacity could decrease, even though the size of the underlying
+  // storage does not decrease.
+  CHECK_NOTHROW(cmat.resize(3, 9));
+  CHECK(cmat.capacity() == 30);
```

Comment:
This change in the semantics of capacity() is no good for other code such as in Walker::registerData(). They are assume to match "std::vector::capacity() returns number of elements that can be held in the allocated storage." This remains 32 even if you have changed n_max and m_max such that you can't index all the allocated storage.  You could resize again and get access to all the allocated storage and the capacity.

The constant size for this object is the max number of elements so that a constant offset to the next element cant be maintained when it is packed into byte buffer and sent over the wire.

My thought on returning n_max_ * m_max_ is that was the area we committed to copying onto the wire.  The actual std::vector::capacity() could always be in blocks of 32 or whatever it just must be  => size().

I think a member variable for the "constant size" i.e. the capacity of the container not the underlying vector is necessary for this change.